llvm/test/SYCLLowerIR/PropagateAspectsUsage/virtual-functions-3.ll

```diff
@@ -0,0 +1,51 @@
+; RUN: opt -passes=sycl-propagate-aspects-usage < %s -S | FileCheck %s
+
+%Foo = type { i32 }
+%Bar = type { i32 }
+
+; CHECK: @vfnFoo() #0 !sycl_used_aspects ![[#aspectsFoo:]]
+define spir_func void @vfnFoo() #0 {
+  call void @subFoo()
+  ret void
+}
+
+define spir_func void @subFoo() {
```

Comment:
should we check the aspects for `subFoo` and `subBar` too?

Reply:
We could, but that part should be already covered by tests for "core" aspects propagation mechanism. Those tests are specially focused on scenarios where a kernel does not use a function directly, but still inherits its aspects, because it uses a "set" where this function belongs to.